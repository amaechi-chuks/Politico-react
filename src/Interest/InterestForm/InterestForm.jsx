/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '../../Components/Global/Buttons';
import '../../Party/PartyCards/partycard.css';

class InterestForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      interestDetails: {
        office: '',
        party: '',
      },
    };
  }

  handleChange = ({ target }) => {
    const { interestDetails } = this.state;
    interestDetails[target.id] = target.value;
    this.setState({ interestDetails });
  };

  handleClick = () => {
    const { declareInterest } = this.props;
    const id = JSON.parse(localStorage.getItem('user')).id;
    declareInterest(id);
  };

  render() {
    const { partyList, officeList } = this.props;
    const { officeList: allOffice } = officeList;
    const { partyList: allParties } = partyList;
    return (
      <div className="main-title">
        <div className="image_title_bio">
          <div className="form-card">
            <label htmlFor="issue-type">
              Select Political Party
              <span>*</span>
            </label>
            <select id="party" onChange={this.handleChange}>
              <option default>-- Select Political Party --</option>
              {allParties.map(party => (
                <option key={party.id}>{party.name}</option>
              ))}
            </select>
            <label htmlFor="issue-type">
              Select Office Type
              <span>*</span>
            </label>
            <select onChange={this.handleChange}>
              <option default>-- Select Office Type --</option>
              {allOffice.map(office => (
                <option key={office.id}>{office.name}</option>
              ))}
            </select>
            <Button
              value="Apply"
              className="btn btn-primary user-form"
              onClick={this.handleClick}
            />
          </div>
        </div>
      </div>
    );
  }
}
InterestForm.defaultProps = {
  officeList: {},
  partyList: {},
};

InterestForm.propTypes = {
  declareInterest: PropTypes.func.isRequired,
  officeList: PropTypes.shape(),
  partyList: PropTypes.shape(),
};
export default InterestForm;
