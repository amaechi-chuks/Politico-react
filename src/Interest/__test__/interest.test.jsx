import React from 'react';
import { shallow } from 'enzyme';
import InterestForm from '../InterestForm/InterestForm';

describe('Interest component', () => {
  it('should match snapshot', () => {
    const wrap = <InterestForm />;
    expect(wrap).toMatchSnapshot();
  });
});
