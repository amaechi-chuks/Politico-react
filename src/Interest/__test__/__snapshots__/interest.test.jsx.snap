// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interest component should match snapshot 1`] = `
<InterestForm
  officeList={Object {}}
  partyList={Object {}}
/>
`;
